rustls/src/client/hs.rs

```diff
@@ -77,21 +77,22 @@ fn find_session(
     sess: &mut ClientSessionImpl,
     dns_name: webpki::DNSNameRef,
 ) -> Option<persist::ClientSessionValue> {
-    let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
-    let key_buf = key.get_encoding();
-
-    let maybe_value = sess
-        .config
-        .session_persistence
-        .get(&key_buf);
+    let key_buf = {
+        let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
+        key.get_encoding()
+    };
```

Comment:
Seems like this change to the scope of `key` is out of scope since it isn't needed for removing the `unwrap()` and it isn't clearly better (or worse) than what was there before. I would revert this change and maybe add this to a TODO for discussing code style changes.

Reply:
This has been reverted.